src/lando/headless_api/tests/test_automation_api.py

```diff
@@ -589,6 +589,109 @@ def test_automation_job_add_commit_fail(
     assert scm.push.call_count == 0
 
 
+PATCH_DIFF = """
+diff --git a/test.txt b/test.txt
+--- a/test.txt
++++ b/test.txt
+@@ -1,1 +1,2 @@
+ TEST
++adding another line
+""".lstrip()
+
+
+@pytest.mark.django_db
+def test_automation_job_create_commit_success_hg(
```

Comment:
This test and the one below can be combined into a parameterized test, as they mostly use the same code. The worker and repos can be abstracted similar to how other tests are abstracted without having to do anything SCM specific in the tests. For an example, see tests in test_landings.py. Basically adding a `get_automation_worker` fixture, since you already are fetching `repo_mc` based on the scm type.

Only the happy path is tested, a test should be added to test the exception handling that is implemented as well.